ejb/WFLY-12190-http-server-to-server-config.adoc

```diff
@@ -0,0 +1,153 @@
+= EJB over HTTP server-to-server invocation
+:email:             tomasz.adamski@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+=== Issue
+* https://issues.jboss.org/browse/WFLY-12190
+
+=== Dev Contacts
+
+* mailto:tadamski@redhat.com
+
+=== QE Contacts
+
+=== Testing By
+[ ] Engineering
+
+[x] QE
+
+=== Affected Projects or Components
+EJBs
+
+== Introduction
+
+PRODMGT-268 (https://issues.redhat.com/browse/PRODMGT-268) added the ability to configure EJB invocations over HTTP protocol. Customers would like to use this feature in server-to-server configurations.
+
+== Proposed configuration
+
+ It was requested to modify server configuration so that http protocol can be used from within remote outbound connections. Sample configuration would be as follows:
+[source]
+----
+<remote-outbound-connection name="http-connection" outbound-socket-binding-ref="binding-remote-ejb-connection" protocol="http"/>
+----
+
+Such connection can be then used from within jboss-ejb-client.xml configuration file. The following issues have to be taken into consideration regarding such configuration:
+
+=== Http config in remoting subystem
+
+Remoting subsystem is responsible for the configuration of connection based on jboss-remoting protocol. The attributes and properties of remote-outbound-connections are used to configure remoting connections and make no sense in for http protocol. Simply adding another protocol to remote-outbound-connection may seem like a simplest solution from the customer point of view, but it introduces incoherent subsystem configuration and in my opinion it is unnaceptable.
+
+=== The need for dynamic discovery
+
+Remoting connections can be configured without any additional deployment information because they rely on dynamic discovery algorithms provided by the EJB client protocol. There is no implementation of http discovery and it has to be designed. I would describe possible discovery algorithms in further part of this analysis.
+
+== Static discovery
+Currently there already is a possiblity to configure server-to-server EJB invocations over HTTP protocol. In order to do this, static discovery has to be used.
+Example of such configuration is as follows:
+
+
+[source]
+----
+<subsystem xmlns="urn:jboss:domain:ejb3:6.0">
+        (...)
+        <remote connector-ref="http-remoting-connector" thread-pool-name="default">
+            <channel-creation-options>
+                <option name="MAX_OUTBOUND_MESSAGES" value="1234" type="remoting"/>
+            </channel-creation-options>
+            <profiles>
+                <profile name="test-profile">
+                    <static-ejb-discovery>
+                        <module uri="http://${server.ip.address}:8180/wildfly-services" module-name="foo"/>
```

Comment:
I'm unfamiliar with the http remoting so what code represents the `module-name` `foo` here? That's the deployment or what? (I'm sorry that I didn't catch it).

Reply:
It's module part of EJB identifier (as used f.e. in JDNI lookup). BTW this is just an example as static discovery implementation/upgrade is not part of this RFE.

Reply:
I see.
Just I have still this doubt - from the design document it seems  that the `static ejb discovery` will be part of the solution. There is section which explains what is it, then this code snippet is used in both code pieces of the document.

The static ejb discovery won't be then configurable at all?

Reply:
static-ejb-discovery is already implemented and works - it was put into the analysis document to show the difference between the new solution (dynamic discovery) and the currently available one

Reply:
oh, ok. my wrong. thanks for clarification.